Ginger/Ginger/ApiModelsFolder/APIModelParamsWizardPage.xaml.cs

```diff
@@ -186,6 +190,33 @@ private void ValueExpressionButton_Click(object sender, RoutedEventArgs e)
             EnhancedActInputValue AIV = (EnhancedActInputValue)xAPIModelParamsValueUCGrid.CurrentItem;
             ValueExpressionEditorPage VEEW = new ValueExpressionEditorPage(AIV, ActInputValue.Fields.Value);
             VEEW.ShowAsWindow();
-        }       
+        }
+        private void MapOutputToDataSource(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                DataSourceTablesListPage dataSourceTablesListPage = new DataSourceTablesListPage();
+                dataSourceTablesListPage.ShowAsWindow();
+
+                if (dataSourceTablesListPage.DSName == "" || dataSourceTablesListPage.DSTableName == "")
+                {
+                    Reporter.ToUser(eUserMsgKeys.MappedtoDataSourceError);
+                    return;
+                }
+                                               
+                foreach (EnhancedActInputValue inputVal in mAddApiModelActionWizardPage.EnhancedInputValueList)
+                {
+                    string sColName = inputVal.Param.Replace("[", "_").Replace("]", "").Replace("{", "").Replace("}", "");
+                    inputVal.Value = "{DS Name=" + dataSourceTablesListPage.DSName + " DST=" + dataSourceTablesListPage.DSTableName + " ACT=MASD MASD=N MR=N IDEN=Cust ICOLVAL=" + sColName + " IROW=NxtAvail}";
+                }
+
+            }
+            catch (System.Exception ex)
+            {
+                Reporter.ToLog(eAppReporterLogLevel.ERROR, ex.StackTrace);
+                Reporter.ToUser(eUserMsgKeys.MappedtoDataSourceError);
```

Comment:
when writing to log need to set the error logic and then only provide ex, exception message/stacktrace/rest will automatically be pulled from it. 
need to write it like below:
Reporter.ToLog(eAppReporterLogLevel.ERROR, "Error occured while mapping the API Model params to Data Source", ex);